app/src/main/java/com/tari/android/wallet/data/repository/TariAddressRepository.kt

```diff
@@ -0,0 +1,53 @@
+package com.tari.android.wallet.data.repository
+
+import com.tari.android.wallet.extension.getResultWithError
+import com.tari.android.wallet.model.TariWalletAddress
+import com.tari.android.wallet.service.TariWalletService
+import com.tari.android.wallet.service.connection.TariWalletServiceConnection
+import com.tari.android.wallet.util.Constants
+import com.tari.android.wallet.util.extractEmojis
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class TariAddressRepository @Inject constructor(
+    private val serviceConnection: TariWalletServiceConnection,
+) {
+
+    suspend fun walletAddressFromHex(hex: String): Result<TariWalletAddress> {
+        return serviceConnection.doOnWalletServiceConnectedWithValue { walletService ->
+            walletService.walletAddressFromHex(hex)
+        }
+    }
+
+    suspend fun walletAddressFromEmojiId(emojiId: String): Result<TariWalletAddress> {
+        return serviceConnection.doOnWalletServiceConnectedWithValue { walletService ->
+            walletService.getResultWithError { error, service -> service.getWalletAddressFromEmojiId(emojiId, error) }
+        }
+    }
+
+    /**
+     * Try to find a valid Emoji ID or HEX in the given text and parse it to a [TariWalletAddress]. Returns null if no valid address is found.
+     */
+    suspend fun parseValidWalletAddress(text: String): TariWalletAddress? {
```

Comment:
The address is now validated by `TariAddress` initialiser.

Reply:
This method is parsing valid addresses from the given text and validating them using FFI lib method. This method is needed for parse QR code text or share text

Reply:
The definition of the Tari Address can change overtime so checking the values on the client side using static, arbitrary values creates a potential point of failure.

Reply:
Yes, but I need to extract it from the text containing not only wallet address